### PR TITLE
Popup: Highlight preferences on search

### LINKED ISSUE #1111: Preference pane search: highlight matching options
### Concept

It would be neat if script options that matched the current search string were subtly visually highlighted. For example, right now when searching for the option to unsticky post avatars, I might type "sticky" into the search box, click to open the options for Tweaks... and then have to read through all of the options to find the one I'm looking for, since the search doesn't descend any further than that.

### PR DESCRIPTION
**Pull request by marcustyphoon**
### Description

Implements a super-simple highlight around preference options that match the current search.

Maybe this should be a different color in dark mode? Probably. I'm not much of a dark mode user, so maybe someone could weigh in on what feels appropriate. (Honestly, I'm pretty meh on the light mode color too.)

Resolves #1111

### CODE REVIEW

**Comment by marcustyphoon** [comment]
(`font-weight: bold` is another option.)

**Review by AprilSylph — CHANGES_REQUESTED** [review]

File: src/browser_action/popup.css [hunk 1]
```diff
@@ -11,6 +11,8 @@
   --accent: 10, 132, 255;
   --passive-grey: 223, 223, 229;
   --active-grey: 207, 207, 216;
+
+  --search-highlight: 255, 255, 0, 0.15;
```
[the remark this answers is not in the document]
Comment (marcustyphoon) [reply src/browser_action/popup.css]: Oh, that's very yellow when used as a highlight at 100% opacity.

**Comment by AprilSylph** [comment]
Hmm, maybe the yellow highlight is too much on dark mode. Firefox highlights matching text when searching for settings with that yellow on light mode, but with `#003eaa` (RGB: `0, 62, 170`) on dark mode.

**Comment by marcustyphoon** [comment]
Ah, now I remember why I had added that line to the dark mode section :D

**Comment by marcustyphoon** [comment]
An option:




```diff
.script li.search-highlighted {
  background: rgb(var(--search-highlight));
+  border-radius: 6px;
}
```

**Review by AprilSylph — APPROVED** [review]

File: src/browser_action/configuration.css [hunk 2]
```diff
@@ -170,7 +170,11 @@ label[for="filter"] {
 .script li {
   display: flex;
   align-items: center;
-  padding: 0.5ch 0;
+  padding: 0.5ch;
+}
+
+.script li.search-highlighted {
+  background: rgb(var(--search-highlight));
```
Comment (AprilSylph) [inline src/browser_action/configuration.css:177]:
```suggestion
  background-color: rgb(var(--search-highlight));
```

**Review by AprilSylph — CHANGES_REQUESTED** [review]

**Comment by AprilSylph** [comment]
Not sure how I feel about the rounded highlight. TBH I want to go all-in on the squareness and push up against the viewport edges but I don't remember how the padding is set up and don't want to either 1) resort to negative margins or 2) accidentally break something

I guess there aren't that many preference types to check and update, but still, I would rather review that as a separate PR.

**Pull request merged**